google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/ConnectionSettings.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.bigquery;
+
+import com.google.auto.value.AutoValue;
+import java.util.List;
+
+/** ConnectionSettings for setting up a BigQuery query connection. */
+@AutoValue
+public abstract class ConnectionSettings {
+  ConnectionSettings() {
+    // Package private so users can't subclass it but AutoValue can.
+  }
+
+  /** Returns the synchronous response timeoutMs associated with this query */
+  public abstract Long getRequestTimeout();
+
+  /** Returns the connection properties for connection string with this query */
+  public abstract List<ConnectionProperty> getConnectionProperties();
+
+  /** Returns the default dataset */
```

Comment:
So, a general comment about all these special methods.  Should they just be connectionProperties, and the client documents specific property keys and how they get used in the client?

Reply:
okay let me use this for things like `defaultDataset`, `maxResult`, etc.